### DISCUSSION TITLE
Converting roadtypes/bridges in towns with poor rating should only be disallowed when downgrading to lower speed

### DISCUSSION THREAD

**Discussion by Andrew350** [OpenTTD/OpenTTD#11739] [Features]
## Version of OpenTTD
20210111-master (since 20210105, see #8457)
## Expected result
Towns allow upgrading roads and bridges to faster/equivalent type, even with poor ratings
## Actual result
Towns always disallow conversion with poor ratings, even though upgrading would ostensibly benefit them. No town council would ever turn down someone else paying to upgrade their infrastructure 😉 Trying to convert to a _slower_ roadtype on the other hand _should_ require good ratings.

Ideally this should be extended to also allow bridge types to be "downgraded" when you have good ratings (current behavior **never** allows downgrading town-owned bridges). This would be more consistent with roadtype behavior, and would make it possible to change the type of bridges in cities without _always_ having to demolish them first in order to build a "lesser" type (as long as you maintain good ratings, anyway).
## Steps to reproduce
Load a roadtype NewGRF and try to convert a road or bridge to a faster type in a town with a poor company rating

**Comment by LC-Zorg** [comment]
1. Would you like to have the motorway outside your window? :)
2. If you were the mayor, would you agree to higher road maintenance costs? And this while doing it in the interest of some company that will run like crazy through the city with its trucks? :D

No, in my opinion, any reconstruction should require the consent of the city. What's more, it should also influence the company's rating in the city (-1 / section?), If only to prevent players from mindlessly rebuilding all streets.

It would be nice if the level of this impact could be determined by additions, because the types of roads can be very different - nobody wants a gravel road or a highway outside the window, but one with nice pavements or a boardwalk shouldn't raise objections. Perhaps even the level of assessment required for consent to redevelopment should be specified. Reconstruction of a dirt road into a gravel road, i.e. a lower category for a higher one, up to a certain level should not have a negative impact on the assessment.

By the way, it would be reasonable to differentiate the settings of local authorities and the impact of actions on the assessment.

**Comment by mattkimber** [comment]
I think there's a fix for roadtype upgrading rules implied here, which happened to us in a recent multiplayer game: one player upgraded their town to "motorway" (the fastest road), on which house building was disabled. They then couldn't downgrade, because with unlimited speed it was considered to be superior to the 30mph roads on which houses could be built.

Regardless, I think bridges and roadtypes should both be consistent, i.e. bridges follow the same upgrade/downgrade rules as roads. There is scope for roadtype conversion to prevent players "upgrading" town-owned roads which enable building houses to ones which don't, as this is a destructive option players can't easily recover from (and enables multiplayer griefing), but I think that's a separate issue.
